src/lib/common/limits.h

```diff
@@ -112,6 +112,8 @@
 #define DEFAULT_OUT_REQ_MSG_MAX_SIZE (8 * 1024 * 1024)  // 8 MB default max size of any outgoing request message (see CLI -outReqMsgMaxSize)
 
 
+#define DEFAULT_THRESHOLD_MAX_SIZE 3
```

Comment:
You are using an absolute value as threshold. However, we could have queues of different sizes, so an absolute threshold is not a good idea. Please, change your implementation to use relative thresholds.

By default, I'd suggest to use "80% of the size of the queue" as default.

Reply:
Fix in https://github.com/telefonicaid/fiware-orion/pull/4375/commits/51a71bbd8113d91b0c1858d8f1f1aada294a61be